modules/environments/azure_devops.tf

```diff
@@ -0,0 +1,38 @@
+resource "azuredevops_serviceendpoint_azurerm" "endpoint" {
+  count                 = var.pipeline_environment == "prod" && var.env != "ptl" && var.env != "ptlsbox" ? 1 : 0
+  project_id            = "PlatformOperations"
+  service_endpoint_name = "OPS-APPROVAL-GATE-${upper(var.env)}-ENVS"
+  credentials {
+    serviceprincipalid  = azuread_service_principal.sp.application_id
+    serviceprincipalkey = azuread_application_password.token.value
+  }
+  azurerm_spn_tenantid      = data.azurerm_client_config.current.tenant_id
+  azurerm_subscription_id   = data.azurerm_subscriptions.sds[0].subscriptions[0].subscription_id
+  azurerm_subscription_name = data.azurerm_subscriptions.sds[0].subscriptions[0].display_name
+}
+
+resource "azuredevops_serviceendpoint_azurerm" "ptl_endpoint" {
+  count                 = var.pipeline_environment == "prod" && var.env == "ptl" ? 1 : 0
```

Comment:
what's with these conditions?

Reply:
This will only work in the prod pipeline because the sds subscriptions are in the common platform tenant so it errors in sandbox - those service connections are currently pointing at sds subscriptions. If there were subscriptions with the same name in the dev tenant then it would work without the conditions.